plugin/pkg/admission/imagepolicy/admission.go

```diff
@@ -0,0 +1,259 @@
+		})
+
+		if err := result.Error(); err != nil {
+			return a.webhookError(err)
+		}
+		var statusCode int
+		if result.StatusCode(&statusCode); statusCode < 200 || statusCode >= 300 {
+			return a.webhookError(fmt.Errorf("Error contacting webhook: %d", statusCode))
+		}
+
+		if err := result.Into(review); err != nil {
+			return a.webhookError(err)
+		}
+
+		a.responseCache.Add(string(cacheKey), review.Status, a.statusTTL(review.Status))
+	}
+
+	if !review.Status.Allowed {
+		return errors.New(review.Status.Reason)
+	}
+
+	return nil
+}
+
+// NewImagePolicyWebhook a new imagePolicyWebhook from the provided config file.
+// The config file is specified by --admission-controller-config-file and has the
+// following format for a webhook:
+//
+//   {
+//     "ImagePolicy.webhook": {
+//        "kubeConfigFile": "path/to/kubeconfig/for/backend",
+//        "allowTTL": 500,          # time in ms to cache approval
+//        "denyTTL": 500,           # time in ms to cache denial
+//        "retryBackoff": 500,      # time in ms to wait between retries
+//        "defaultAllow": true      # determines behavior if the webhook backend fails
+//     }
+//   }
+//
+// The kubeconfig property refers to another file in the kubeconfig format which
+// specifies how to connect to the webhook backend.
+//
+// The kubeconfig's cluster field is used to refer to the remote service, user refers to the returned authorizer.
+//
+//     # clusters refers to the remote service.
+//     clusters:
+//     - name: name-of-remote-imagepolicy-service
+//       cluster:
+//         certificate-authority: /path/to/ca.pem      # CA for verifying the remote service.
+//         server: https://images.example.com/policy # URL of remote service to query. Must use 'https'.
+//
+//     # users refers to the API server's webhook configuration.
+//     users:
+//     - name: name-of-api-server
+//       user:
+//         client-certificate: /path/to/cert.pem # cert for the webhook plugin to use
+//         client-key: /path/to/key.pem          # key matching the cert
+//
+// For additional HTTP configuration, refer to the kubeconfig documentation
+// http://kubernetes.io/v1.1/docs/user-guide/kubeconfig-file.html.
+func NewImagePolicyWebhook(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+	configFile, err := ioutil.ReadAll(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return newWithBackoff(configFile, defaultRetryBackoff)
+}
+
+// newWithBackoff allows tests to skip the sleep.
+func newWithBackoff(configFile []byte, initialBackoff time.Duration) (*imagePolicyWebhook, error) {
+	var config AdmissionConfig
+	err := json.Unmarshal(configFile, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	whConfig := config.ImagePolicyWebhook
+	if whConfig.RetryBackoff == 0 {
+		whConfig.RetryBackoff = initialBackoff
+	}
+
+	// convert to ms; unmarshalling gives ns
+	whConfig.RetryBackoff *= time.Millisecond
```

Comment:
Validate some bounds here to avoid big mistakes based on units.  Something unreasonably large for purpose to catch unit mistakes. 

Anything less than ten seconds and anything larger than 30 minutes?  You could have a magic value (zero probably) for "no caching"